Library/Formula/devil.rb

```diff
@@ -1,10 +1,39 @@
 class Devil < Formula
   desc "Cross-platform image library"
   homepage "https://sourceforge.net/projects/openil/"
-  url "https://downloads.sourceforge.net/project/openil/DevIL/1.7.8/DevIL-1.7.8.tar.gz"
-  sha256 "682ffa3fc894686156337b8ce473c954bf3f4fb0f3ecac159c73db632d28a8fd"
   revision 1
 
+  stable do
+    url "https://downloads.sourceforge.net/project/openil/DevIL/1.7.8/DevIL-1.7.8.tar.gz"
+    sha256 "682ffa3fc894686156337b8ce473c954bf3f4fb0f3ecac159c73db632d28a8fd"
+
+    # fix compilation issue for ilur.c
+    patch do
+      url "https://raw.githubusercontent.com/Homebrew/patches/3db2f9727cea4a51fbcfae742518c614020fb8f2/devil/patch-src-ILU-ilur-ilur.c.diff"
+      sha256 "ce96bc4aad940b80bc918180d6948595ee72624ae925886b1b770f2a7be8a2f9"
+    end
+  end
+
+  head do
+    url "https://github.com/DentonW/DevIL.git"
+
+    option "with-ilut", "Also build the ILUT library"
+
+    depends_on "libtool" => :build
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+
+    if build.with? "ilut"
+      depends_on "freeglut" => :run
```

Comment:
This was fine how you had it earlier as far as I saw:
```ruby
depends_on "homebrew/x11/freeglut" => :optional
```
Or:
```ruby
depends_on "homebrew/x11/freeglut" if build.with? "ilut"
```
We'd prefer the first style really. The `if build.with?` style dependencies don't declare themselves entirely obviously to users.